service/src/main/java/bio/terra/profile/service/profile/flight/create/CreateProfileVerifyDeployedApplicationStep.java

```diff
@@ -35,7 +35,7 @@ public StepResult doStep(FlightContext context) {
                           Objects.equals(
                                   app.getManagedResourceGroupId(),
                                   profile.getRequiredManagedResourceGroupId())
-                              && app.getSubscriptionId() == profile.getRequiredSubscriptionId())
+                              && Objects.equals(app.getSubscriptionId(), profile.getRequiredSubscriptionId()))
```

Comment:
We were getting away with a bad object reference comparison; we want an bit-for-bit object value comparison instead. This is working today because I believe the getManagedAppDeployments method is just reflecting the UUID object back in it's return value.